axelor-quality/src/main/java/com/axelor/apps/quality/service/QIAnalysisServiceImpl.java

```diff
@@ -35,4 +97,168 @@ public int setAdvancement(QIAnalysis qiAnalysis) {
+    createQIActionDistributionForOthers(
+        qiAnalysis,
+        recepient,
+        recepientPartner,
+        qualityConfig,
+        qiActionDistributionSequence,
+        generatedQiActionDistributionListForOthers);
+    return generatedQiActionDistributionListForOthers;
+  }
+
+  @Transactional(rollbackOn = Exception.class)
+  protected void createQIActionDistribution(
+      QIAnalysis qiAnalysis,
+      QualityConfig qualityConfig,
+      Sequence qiActionDistributionSequence,
+      List<QIActionDistribution> generatedQiActionDistributionList,
+      Partner responsiblePartner,
+      List<QITask> qiTasks)
+      throws AxelorException, FileNotFoundException, IOException {
+    QIActionDistribution qiActionDistribution = new QIActionDistribution();
+    qiActionDistribution.setSequence(
+        sequenceService.getSequenceNumber(
+            qiActionDistributionSequence,
+            QIActionDistribution.class,
+            "sequence",
+            qiActionDistribution));
+    qiActionDistribution.setRecipient(qiTasks.get(0).getResponsible());
+    qiActionDistribution.setRecipientPartner(responsiblePartner);
+    qiActionDistribution.setQiDecision(qiTasks.get(0).getQiDecision());
+    qiActionDistribution.setQiTaskSet(new HashSet<>(qiTasks));
+    qiActionDistributionRepository.save(qiActionDistribution);
+    generatedQiActionDistributionList.add(qiActionDistribution);
+    qiAnalysis.addQiActionDistributionListItem(qiActionDistribution);
+    qiActionDistribution.setGeneratedFile(getGeneratedFile(qualityConfig, qiActionDistribution));
+  }
+
+  @Transactional(rollbackOn = Exception.class)
+  protected void createQIActionDistributionForOthers(
+      QIAnalysis qiAnalysis,
+      Integer recepient,
+      Partner recepientPartner,
+      QualityConfig qualityConfig,
+      Sequence qiActionDistributionSequence,
+      List<QIActionDistribution> generatedQiActionDistributionListForOthers)
+      throws AxelorException, FileNotFoundException, IOException {
+    QIActionDistribution qiActionDistribution = new QIActionDistribution();
+    qiActionDistribution.setSequence(
+        sequenceService.getSequenceNumber(
+            qiActionDistributionSequence,
+            QIActionDistribution.class,
+            "sequence",
+            qiActionDistribution));
+    qiActionDistribution.setRecipient(recepient);
+    qiActionDistribution.setRecipientPartner(recepientPartner);
+    qiActionDistributionRepository.save(qiActionDistribution);
+    generatedQiActionDistributionListForOthers.add(qiActionDistribution);
+    qiAnalysis.addQiActionDistributionListItem(qiActionDistribution);
+    qiActionDistribution.setGeneratedFile(getGeneratedFile(qualityConfig, qiActionDistribution));
+  }
+
+  protected DMSFile getGeneratedFile(
+      QualityConfig qualityConfig, QIActionDistribution qiActionDistribution)
+      throws AxelorException, FileNotFoundException, IOException {
+    BirtTemplate qiActionDistributionBirtTemplate =
```

Comment:
Here you can use the new service to get the birt template (therefore add new a method), then you can remove qualityConfig from parameter of method